fast64_internal/oot/oot_constants.py

```diff
@@ -44,449 +52,24 @@
 	("Night", "Night", "Night")
 ]
 
-ootEnumTransitionActorID = [
-	("Custom", "Custom", "Custom"),
-	("ACTOR_EN_DOOR", "EN_DOOR", "EN_DOOR"),
-	("ACTOR_DOOR_SHUTTER", "DOOR_SHUTTER", "DOOR_SHUTTER"),
-	("ACTOR_DOOR_WARP1", "DOOR_WARP1", "DOOR_WARP1"),
-	("ACTOR_DOOR_TOKI", "DOOR_TOKI", "DOOR_TOKI"),
-	("ACTOR_DOOR_ANA", "DOOR_ANA", "DOOR_ANA"),
-	("ACTOR_DOOR_GERUDO", "DOOR_GERUDO", "DOOR_GERUDO"),
-	("ACTOR_DOOR_KILLER", "DOOR_KILLER", "DOOR_KILLER"),
-]
+# Add every actor that is a transition actor (defined by the category 'ACTORCAT_DOOR')
+ootEnumTransitionActorID = [(actorNode.get('ID'), actorNode.get('Name'), \
```

Comment:
`actorNode.get('ID')` is the enum name (like `ACTOR_EN_DOOR`)
if you look at the blender api for EnumProperty, this as the first element of the tuple is what is actually saved in the blend
we can't change it as we need to keep compatibility with old blends, but it's something to keep in mind for when the enum names may change.

this means the xml should probably use "ID" or something else as something that will never ever change, and have another "EnumName" thing for the enum name, which may change. and currently both would be the same

Reply:
I don't get it, like older blends shouldn't have like ``ACTOR_EN_BOX`` as a transition actor right? letting the user choose from the whole list was a mistake as it's just odd to be able to choose actors that don't fall under the actor category "doors"

and the value only matters when you upgrade, so I don't really understand

Reply:
in practice, imagine decomp changes `ACTOR_EN_DOOR` to `ACTOR_DOOR`
the xml would change `ID="ACTOR_EN_DOOR"` to `ID="ACTOR_DOOR"` right
but this ID value is being used as the "identifier" https://docs.blender.org/api/current/bpy.props.html#bpy.props.EnumProperty which is used to save the property in the .blend
so changing the ID would break old blends

Reply:
how about using the actor keys (like 0x000A for en_box) instead? this should never change

Reply:
what about the existing blends though
and ideally we wouldn't want to hardcode the hex values, they may change in decomp mods

Reply:
ok then keeping ``ID`` as the current decomp IDs and replace ``Key`` by what's been used as "ID" currently and never changing that ever (and remove the ``ACTOR_`` part of course)

Reply:
(I just don't feel like finding a proper name for all of the ~430 actors in the game lol)